stacker_blueprints/empire/daemon.py

```diff
@@ -0,0 +1,604 @@
+        self.create_load_balancer()
+        self.create_ecs_resources()
+        self.create_log_group()
+
+    def create_conditions(self):
+        t = self.template
+        ssl_condition = Not(Equals(Ref("ELBCertName"), ""))
+        t.add_condition("UseSSL", ssl_condition)
+        self.template.add_condition(
+            "UseIAMCert",
+            Not(Equals(Ref("ELBCertType"), "acm")))
+        t.add_condition(
+            "EnableSNSEvents",
+            Equals(Ref("EventsBackend"), "sns"))
+        t.add_condition(
+            "CreateSNSTopic",
+            And(Equals(Ref("EventsSNSTopicName"), ""),
+                Condition("EnableSNSEvents")))
+        t.add_condition(
+            "EnableCloudwatchLogs",
+            Equals(Ref("RunLogsBackend"), "cloudwatch"))
+        t.add_condition(
+            "CreateRunLogsGroup",
+            And(Equals(Ref("RunLogsCloudwatchGroup"), ""),
+                Condition("EnableCloudwatchLogs")))
+        t.add_condition(
+            "EnableAppEventStream",
+            Equals(Ref("LogsStreamer"), "kinesis"))
+
+    def create_security_groups(self):
+        t = self.template
+
+        t.add_resource(
+            ec2.SecurityGroup(
+                ELB_SG_NAME,
+                GroupDescription="Security group for load balancer",
+                VpcId=Ref("VpcId")))
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "ELBPort80FromTrustedNetwork",
+                IpProtocol="tcp", FromPort="80", ToPort="80",
+                CidrIp=Ref("TrustedNetwork"),
+                GroupId=Ref(ELB_SG_NAME)))
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "ELBPort443FromTrustedNetwork",
+                IpProtocol="tcp", FromPort="443", ToPort="443",
+                CidrIp=Ref("TrustedNetwork"),
+                GroupId=Ref(ELB_SG_NAME)))
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "ELBPort443GitHub",
+                IpProtocol="tcp", FromPort="443", ToPort="443",
+                CidrIp=Ref("GitHubCIDR"),
+                GroupId=Ref(ELB_SG_NAME)))
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "80ToControllerPort8081",
+                IpProtocol="tcp", FromPort="8081", ToPort="8081",
+                SourceSecurityGroupId=Ref(ELB_SG_NAME),
+                GroupId=Ref("InstanceSecurityGroup")))
+
+    def create_custom_cloudformation_resources(self):
+        t = self.template
+
+        queue = sqs.Queue("CustomResourcesQueue")
+        topic = sns.Topic(
+            "CustomResourcesTopic",
+            Subscription=[sns.Subscription(
+                Protocol="sqs",
+                Endpoint=GetAtt("CustomResourcesQueue", "Arn"))])
+        queue_policy = sqs.QueuePolicy(
+            "CustomResourcesQueuePolicy",
+            Queues=[Ref(queue)],
+            PolicyDocument=sns_to_sqs_policy(Ref(topic)))
+
+        t.add_resource(queue)
```

Comment:
actually, this is used because of the ref to `Queues=[Ref(queue)]` and `Ref(topic)` being passed to `sns_to_sqs_policy`